xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -909,6 +909,9 @@ void CVideoPlayer::CloseDemuxer()
 
 void CVideoPlayer::OpenDefaultStreams(bool reset)
 {
+  m_processInfo->ResetAudioCodecInfo();
+  m_processInfo->ResetVideoCodecInfo();
```

Comment:
that won't work. OpenDefaultStream can be called often but won't do anything if parameters did not change.

better explicitly reset if there is no stream of a particular type